src/components/PeopleList.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { Person } from '../types';
+import { PeopleItem } from './PeopleItem';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  people: Person[],
+  peopleFromSesver: Person[],
+};
+
+export const PeopleList: React.FC<Props> = ({
+  people,
+  peopleFromSesver,
+}) => {
+  const [sortPeople, setSortPeople] = useState<Person[]>(people);
+  const [searchParams] = useSearchParams();
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const getSearchSort = useCallback((value: string) => {
+    if (sort !== value) {
+      return { sort: value, order: null };
+    }
+
+    return (!order)
+      ? { sort: value, order: 'desc' }
+      : { sort: null, order: null };
+  }, [sort, order]);
+
+  const handlerSortPeople = useCallback((
+    sortParam: 'name' | 'sex' | 'born' | 'died',
+  ) => {
+    return [...people].sort((a, b) => {
+      if (typeof a[sortParam] === 'string') {
+        const personA = a[sortParam].toString();
+        const personB = b[sortParam].toString();
+
+        return personA.localeCompare(personB);
+      }
+
+      const personA = +a[sortParam];
+      const personB = +b[sortParam];
+
+      return personA - personB;
+    });
+  }, [people]);
+
+  const getSortedPeople = useCallback(() => {
+    let sortedPeople = [...people];
+
+    if (sort) {
+      switch (sort) {
```

Comment:
you don't need it
every case will be covered with this one line: `sortedPeople = handlerSortPeople(sort);`

Reply:
agree, please remove the switch case block